Sources/DistributedActors/_ActorShell.swift

```diff
@@ -159,7 +159,7 @@ public final class _ActorShell<Message: ActorMessage>: _ActorContext<Message>, A
         self._log = Logger.make(system.log, path: address.path)
 
         self.supervisor = _Supervision.supervisorFor(system, initialBehavior: behavior, props: props.supervision)
-        self._deathWatch = DeathWatchImpl(nodeDeathWatcher: system._nodeDeathWatcher ?? system.deadLetters.adapted())
+        self._deathWatch = DeathWatchImpl(nodeDeathWatcher: system._nodeDeathWatcherStore.load()?.value ?? system.deadLetters.adapted())
```

Comment:
As far as I can tell this is actually acceptable for the value not to be set yet here because the type where this is indirectly used (`LocalActorRefProvider`) is invoked before the set in the init dance.

Reply:
Hmmm... the shell always needs the deathwatch, I believe... I'll double check this; perhaps the only actor we start so early is one which does never deathwatch so it's fine... 🤔 

I'll double check but this PR is fine!